docs/src/operate/c8y/connect-mqtt-service.md

```diff
@@ -0,0 +1,120 @@
+
+:::caution
+Cumulocity MQTT service is still a work-in-progress and the external interfaces are bound to change in future.
+The %%te%% interfaces might also change accordingly.
+:::
+
+### Configure the device
+
+Most of the configurations used to connect to the MQTT service endpoint of Cumulocity are same as
+the ones used to connect to the core MQTT endpoint.
+
+1. Configure Cumulocity tenant URL:
+
+   <UserContext>
+   
+   ```sh
+   sudo tedge config set c8y.url $C8Y_URL
+   ```
+   
+   </UserContext>
+
+   :::note
+   Though you're setting the `c8y.url` config, the `c8y.mqtt_service.url` config is used under-the-hood for the connection,
+   as this config is derived from the `c8y.mqtt` config, which is further derived from `c8y.url`, by default.
+   For example, when `c8y.url` is `example.cumulocity.com`, `c8y.mqtt` would be `example.cumulocity.com:8883`
+   and `c8y.mqtt_service.url` would be `example.cumulocity.com:9883`.
+   If the MQTT service url is different from the one that would be derived from `c8y.url` or `c8y.mqtt`,
+   set`c8y.mqtt_service.url` explicitly.
+   :::
+
+1. If you're using device certificates to connect to Cumulocity, set the tenant id (skip this when using username/password):
+
+   <UserContext>
+   
+   ```sh
+   sudo tedge config set c8y.tenant_id $C8Y_TENANT_ID
+   ```
+   
+   </UserContext>
+
+1. Enable connection to MQTT service endpoint:
+
+   ```sh
+   sudo tedge config set c8y.mqtt_service.enabled true
+   ```
+
+1. Provide a topic to subscribe to (only a single topic is allowed at the moment):
+
+   ```sh
+   sudo tedge config set c8y.mqtt_service.topics demo/topic
+   ```
+
+### Configure the cloud to trust the device certificate
+
+Follow the steps [here](./connect.md#making-the-cloud-trust-the-device) to make Cumulocity trust the device certificate.
+
+### Connect the device
+
+The device is connected to MQTT service endpoint along with the core MQTT connection to Cumulocity:
+
+```sh
+sudo tedge connect c8y
+```
+
+This step establishes the bridge connection to both the core endpoint as well as the mqtt service endpoints simultaneously.
+
+### Validate the connection
+
+Once connected, all messages published to `c8y-mqtt/#` topics are forwarded to the MQTT service endpoint.
+
+To validate the same, connect a different MQTT client directly to the MQTT service endpoint (`mosquitto_sub` in this example)
+and subscribe to the desired topic:
+
+<UserContext>
+
+```sh
+mosquitto_sub -d -v -h $C8Y_URL -p 2883 -i test-client-123 -u $C8Y_TENANT_ID/$C8Y_USER -P $C8Y_PASSWORD -t test/topic
+```
+
+</UserContext>
+
+:::note
+The test client must use a client id that is different from the device id.
+:::
+
+Once subscribed, publish to the same topic from the %%te%% device:
+
+```sh
+tedge mqtt pub c8y-mqtt/test/topic "hello world"
+```
+
+Now, validate that the same message is received by the test client.
```

Comment:
We're going to need to remove this section, as device isolation is starting to be rolled out this week, which will invalidate these instructions...and given that there won't be an easy way to validate this for now, it's better to have it removed.